rpc/service.c

```diff
@@ -192,15 +192,67 @@ m0_rpc_service_reverse_session_get(struct m0_reqh_service   *service,
 	return M0_RC(rc);
 }
 
+static void rev_conn_free(struct m0_sm_group *grp, struct m0_sm_ast *ast)
+{
+	struct m0_reverse_connection *revc = ast->sa_datum;
+
+	M0_ENTRY("revc=%p", revc);
+	m0_rpc_link_fini(&revc->rcf_rlink);
+	rev_conn_tlink_del_fini(revc);
+	m0_free(revc);
+	M0_LEAVE("revc=%p", revc);
+}
+
+static bool rev_conn_disconnected_cb(struct m0_clink *link)
+{
+	struct m0_reverse_connection *revc;
+
+	revc = container_of(link, struct m0_reverse_connection, rcf_disc_wait);
+	revc->rcf_free_ast = (struct m0_sm_ast){
+		.sa_cb    = &rev_conn_free,
+		.sa_datum = revc,
+	};
+	m0_sm_ast_post(m0_locality0_get()->lo_grp, &revc->rcf_free_ast);
+	return true;
+}
+
+M0_INTERNAL void
+m0_rpc_service_reverse_session_put(struct m0_rpc_session *sess)
```

Comment:
[Question]
When and where is this function called from?

Reply:
```
$ gg m0_rpc_service_reverse_session_put
rm/rm.c:1395:           m0_rpc_service_reverse_session_put(sess);
```
at `remote_free()` when the debtor has no more loans and no more pending incoming requests for them.

Reply:
Thanks